funnel/views/schedule.py

```diff
@@ -259,8 +265,10 @@ def schedule_edit(profile, project):
         'scheduled': session_data(project.sessions, timezone=project.timezone, with_modal_url='edit', with_delete_url=True)
         }
     # Set the proper range for the calendar to allow for date changes
-    first_session = Session.query.filter_by(project=project).order_by("created_at asc").first()
-    last_session = Session.query.filter_by(project=project).order_by("created_at desc").first()
+    first_session = Session.query.filter(
+        Session.start.isnot(None), Session.end.isnot(None)).filter_by(project=project).order_by("created_at asc").first()
+    last_session = Session.query.filter(
+        Session.start.isnot(None), Session.end.isnot(None)).filter_by(project=project).order_by("created_at desc").first()
```

Comment:
Make `scheduled` a `hybrid_property` and you can rewrite this as `Session.query.filter(Session.scheduled)…`